modules/nf-core/deseq2/differential/main.nf

```diff
@@ -14,24 +14,26 @@ process DESEQ2_DIFFERENTIAL {
     tuple val(meta4), path(transcript_lengths_file)
 
     output:
-    tuple val(meta), path("*.deseq2.results.tsv")              , emit: results
-    tuple val(meta), path("*.deseq2.dispersion.png")           , emit: dispersion_plot
-    tuple val(meta), path("*.dds.rld.rds")                     , emit: rdata
-    tuple val(meta), path("*.deseq2.sizefactors.tsv")          , emit: size_factors
-    tuple val(meta), path("*.normalised_counts.tsv")           , emit: normalised_counts
-    tuple val(meta), path("*.rlog.tsv")                        , optional: true, emit: rlog_counts
-    tuple val(meta), path("*.vst.tsv")                         , optional: true, emit: vst_counts
-    tuple val(meta), path("*.deseq2.model.txt")                , emit: model
-    tuple val(meta), path("*.R_sessionInfo.log")               , emit: session_info
+    tuple val(out_meta), path("*.deseq2.results.tsv")              , emit: results
+    tuple val(out_meta), path("*.deseq2.dispersion.png")           , emit: dispersion_plot
+    tuple val(out_meta), path("*.dds.rld.rds")                     , emit: rdata
+    tuple val(out_meta), path("*.deseq2.sizefactors.tsv")          , emit: size_factors
+    tuple val(out_meta), path("*.normalised_counts.tsv")           , emit: normalised_counts
+    tuple val(out_meta), path("*.rlog.tsv")                        , optional: true, emit: rlog_counts
+    tuple val(out_meta), path("*.vst.tsv")                         , optional: true, emit: vst_counts
+    tuple val(out_meta), path("*.deseq2.model.txt")                , emit: model
+    tuple val(out_meta), path("*.R_sessionInfo.log")               , emit: session_info
     path "versions.yml"                                        , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
+    out_meta = meta + meta2
```

Comment:
what is the content of `meta` and `meta2`? It's not clear from the `meta.yml`

Reply:
so `meta` is the meta for contrast channel, that has information like the contrast id, and so on.
Then the `meta2` has the meta information about the count data, experiment id, etc.
Then `meta3` and `meta4` are in this case the same as `meta2`.
But I agree that it is not clear...

Reply:
So the meta.yml says for `meta`:

```
          Groovy Map containing contrast information. This can be used at the
          workflow level to pass optional parameters to the module, e.g.
          [ id:'contrast1', blocking:'patient' ] passed in as ext.args like:
          '--blocking_variable $meta.blocking'.
```

and for meta2:

```
          Groovy map containing study-wide metadata related to the sample sheet
          and matrix
```

I felt that that was pretty descriptive when I wrote it, but happy to revise if you have suggestions.

Reply:
Thanks for the explanation :) `meta` is more clear, but I was wondering the exact keys for `meta2`, for example if it also contains an `id` but with a different value, because then it's unclear which one will have preference if we combine them.

Reply:
Does it mean that meta2.id = experiment id will override meta.id = contrast id?
Probably that is fine as far as we define more unique meta values (id, contrast_id, instead of id for both cases) at the pipeline level to avoid overwritting?

Reply:
Maybe we should create a composite ID of the abundance and contrast IDs

Reply:
OK, changed my mind, I don't think we should merge meta in the modules, it makes 're-joining' results at the workflow level fiddly. 

So I've put some meta handling in the subworkflow instead (without imposing assumptions)